tests/e2e/replication/show_while_creating_invalid_state.py

```diff
@@ -403,5 +440,1016 @@ def check_roles():
+        data_directory_replica.cleanup()
+
+
+def test_sync_replication_when_main_is_killed():
+    # Goal of the test is to check that when main is randomly killed:
+    # -the SYNC replica always contains the exact data that was in main.
+    # We run the test 20 times, it should never fail.
+
+    # 0/ Start main and replica.
+    # 1/ Register replica.
+    # 2/ Insert data in main, and randomly kill it.
+    # 3/ Check that the SYNC replica has exactly the same data than main.
+
+    for test_repetition in range(20):
+        # 0/
+        data_directory_main = tempfile.TemporaryDirectory()
+        data_directory_replica = tempfile.TemporaryDirectory()
+        CONFIGURATION = {
+            "sync_replica": {
+                "args": ["--bolt-port", "7688", "--log-level=TRACE"],
+                "log_file": "sync_replica.log",
+                "setup_queries": ["SET REPLICATION ROLE TO REPLICA WITH PORT 10001;"],
+                "data_directory": f"{data_directory_replica.name}",
+            },
+            "main": {
+                "args": ["--bolt-port", "7687", "--log-level=TRACE", "--storage-recover-on-startup=true"],
+                "log_file": "main.log",
+                "setup_queries": [],
+                "data_directory": f"{data_directory_main.name}",
+            },
+        }
+
+        interactive_mg_runner.start_all(CONFIGURATION)
+
+        # 1/
+        interactive_mg_runner.MEMGRAPH_INSTANCES["main"].query(
+            "REGISTER REPLICA sync_replica SYNC TO '127.0.0.1:10001';"
+        )
+
+        # 2/
+        QUERY_TO_CHECK = "MATCH (n) RETURN COLLECT(n.name);"
+        last_result_from_main = interactive_mg_runner.MEMGRAPH_INSTANCES["main"].query(QUERY_TO_CHECK)[0][0]
+        for index in range(50):
+            interactive_mg_runner.MEMGRAPH_INSTANCES["main"].query(f"CREATE (p:Number {{name:{index}}})")
+            last_result_from_main = interactive_mg_runner.MEMGRAPH_INSTANCES["main"].query(QUERY_TO_CHECK)[0][0]
+            if random.randint(0, 100) > 95:
+                interactive_mg_runner.kill(CONFIGURATION, "main")
+                break
+
+        # 3/
+        # The SYNC replica should have exactly the same data than main.
+        res_from_sync_replica = interactive_mg_runner.MEMGRAPH_INSTANCES["sync_replica"].query(QUERY_TO_CHECK)[0][0]
+        assert last_result_from_main == res_from_sync_replica
+
+        data_directory_main.cleanup()
+        data_directory_replica.cleanup()
+
+
+def test_attempt_to_write_data_on_main_when_async_replica_is_down():
+    # Goal of this test is to check that main can write new data if an async replica is down.
+    # 0/ Start main and async replicas.
+    # 1/ Check status of replicas.
+    # 2/ Add some nodes to main and check it is propagated to the async_replicas.
+    # 3/ Kill an async replica.
+    # 4/ Try to add some data to main.
+    # 5/ Check the status of replicsa.
+    # 6/ Check that the data was added to main and remaining replica.
```

Comment:
```suggestion
    # 5/ Check the status of replicas.
```

Reply:
Use search and replace as there are other accoruences also.

Reply:
Ahh you meant the typo `replicsa` instead of `replicas` I assume!
Corrected.